### PR TITLE
Refactor request decorator based on reek feedback

### LINKED ISSUE #4180: Run Reek on RequestDecorator Class
### What maintenance needs to be done?
 The reek score of this file is currently 14. 

### Level of urgency
- [ ] High
- [ ] Moderate
- [ ] Low

### Why is this maintenance needed?
- The file has a very high complexity score.

### Acceptance criteria
 - [x] Warnings are lowered to 5 or less. 

### Implementation notes, if any
```reek app/models/requests/request_decorator.rb``` to see current report.

### PR DESCRIPTION
**Pull request by maxkadel**
Closes #4180

### CODE REVIEW

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel 🦄

File: app/helpers/requests/application_helper.rb [hunk 1]
```diff
@@ -207,12 +207,12 @@ def item_checkbox(requestable, single_item_form)
     end
 
     ## If any requestable items have a temp location assume everything at the holding is in a temp loc?
-    def current_location_label(mfhd_label, requestable_list)
+    def current_location_label(request_location_label, requestable_list)
```
Comment (christinach) [inline app/helpers/requests/application_helper.rb:210]: can we change the request_location_label to a holding_location_label?

**Review by christinach — CHANGES_REQUESTED** [review]

File: app/models/requests/requestable.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def initialize(bib:, holding: nil, item: nil, location: nil, patron:)
     end
 
     delegate :pick_up_location_id, :pick_up_location_code, :item_type, :enum_value, :cron_value, :item_data?,
-             :temp_loc?, :in_resource_sharing?, :on_reserve?, :inaccessible?, :hold_request?, :enumerated?, :item_type_non_circulate?, :partner_holding?,
+             :temp_loc_other_than_resource_sharing?, :on_reserve?, :inaccessible?, :hold_request?, :enumerated?, :item_type_non_circulate?, :partner_holding?,
```
Comment (christinach) [inline app/models/requests/requestable.rb:43]: I'm a little bit confused with this replacement.
`temp_loc? && !in_resource_sharing?`

Comment (maxkadel) [reply app/models/requests/requestable.rb:43]: For me it took a long time to understand what `temp_loc? && !in_resource_sharing?` was representing, and I found it confusing that neither is used by itself, but that same chunk was in several places. Also reek was complaining about how much `temp_loc?` was being repeated

**Review by christinach — APPROVED** [review]
thanks @maxkadel

**Pull request merged**